Try to fix build on runtime 25.08

user1-github opened the pull request:
The build was failing in the previous PR probably because 25.08 ships GCC 15 which defaults to newer C standard. Forcing it to use an older C standard via CFLAGS might fix this.

user1-github committed:
Try to fix build on runtime 25.08

AsciiWolf commented:
Tested and works great (on x86_64)! Thanks a lot for working on this, @user1-github!

P.S. Any chance you could also take a look at the [SuperTuxKart update](https://github.com/flathub/net.supertuxkart.SuperTuxKart/pull/41)? :-)

user1-github committed:
Switch to -std=gnu17

Slightly newer, but should still work with Xonotic 0.8.6, as I've seen some current Xonotic distro packages force -std=gnu17 to make it build on GCC 15.

user1-github committed:
Change to id, as app-id is deprecated and try building without LTO workaround

Let's see if the workaround is still needed or not with runtime 25.08.

user1-github committed:
LTO workaround is still needed, add it back

user1-github commented:
Tested the last build, works well.
I think this can now be merged.

AsciiWolf commented:
@bones-was-here Please take a look. Thanks! :-)

bones-was-here commented:
That workaround looks ok to me (has a comment! :)  

NB: if this switches from standard sdl2 to sdl2-compat+sdl3 (I can't tell if it does just from https://discourse.flathub.org/t/freedesktop-sdk-25-08-0-released/10399) it may cause issues for nvidia+wayland users because the default will change from xwayland to native, and nvidia still has wayland issues (especially on old GPUs that will never get decent wayland support).  
See: https://gitlab.com/xonotic/darkplaces/-/issues/434#note_2594340983

user1-github commented:
25.08 uses sdl2-compat+sdl3, but AFAIR, Xonotic Flatpak has been defaulting to Wayland since about 2-3 years ago anyway (by using --socket=wayland and --socket=fallback-x11 permissions), so technically updating to runtime 25.08 doesn't change anything in this sense (maybe with some improvement in native Wayland support due to support for recent Wayland protocols in sdl3).

AsciiWolf commented:
So... Are we ready to merge this? :)

bones-was-here reviewed (approved):
Hopefully this is equivalent to the MR approve button
